test/e2e/v3/plugin_cluster_test.go

```diff
@@ -398,43 +380,49 @@ func curlMetrics(kbc *utils.TestContext) string {
 	return metricsOutput
 }
 
-// TODO: Remove this when and other related functions when
-// tests have been changed to use a better implementation.
-// ServiceAccountToken creates the ServiceAccount token Secret.
-// This is to be used when Kubernetes cluster version is >= 1.24.
-// In k8s 1.24+ a ServiceAccount token Secret is no longer
-// automatically generated
-func ServiceAccountToken(kbc *utils.TestContext) {
-	// As of Kubernetes 1.24 a ServiceAccount no longer has a ServiceAccount token
-	// secret autogenerated. We have to create it manually here
+// ServiceAccountToken performs a kubectl create token to request a new Token
+// for the service account in question and use that token for performing the
+// curl command calls as required. Please keep in mind that the output of
+// kubectl create token is not base64 encoded. It is the JWT token itself
+func ServiceAccountToken(kbc *utils.TestContext) (out string, err error) {
 	By("Creating the ServiceAccount token")
-	secretFile, err := createServiceAccountToken(kbc.Kubectl.ServiceAccount, kbc.Dir)
-	Expect(err).NotTo(HaveOccurred())
-	Eventually(func() error {
+	if srvVer := kbc.K8sVersion.ServerVersion; srvVer.GetMajorInt() == 1 && srvVer.GetMinorInt() >= 24 {
```

Comment:
Is there a reason the TokenRequest only works on k8s 1.24+?

I was under the impression that this should work on all the Kubernetes versions we are using because the feature was in beta status as of k8s 1.12 which is earlier than any of the versions that we are testing. I think it could potentially be related to it being disabled within the KinD node we are using but I would be curious what the earliest version we could start using the TokenRequest API in is.

I would be interested to see what happens when slowly backing the minor version down from 24 to as low as we can get it.

Reply:
The reason I put this check in is using the `kubectl create token` command. That is present only in 1.24.x+.

I am trying out a change to use the `TokenRequest` API call for all of them and avoid this check. Will push the changes in a while once locally tested.